DEVX-2070: Changes prior to building and publishing new 6.0 based datagen images

rspurgeon opened the pull request:
## Problem
Want to build a new version of the connector off of the 6.0 CP libraries

## Solution
Update pom.xml and execute a release building 0.3.4 datagen version

<!--- Mark x in the box. -->
##### Does this solution apply anywhere else?
- [ ] yes
- [X] no

## Test Strategy
No code changes in the datagen code with this change, only the base libraries have been upgraded to 6.0.  Unit tests pass

## Release Plan
After building the full datagen docker imags, will validate new images in Confluent Integration Arch demos and examples.

ybyzek reviewed (approved):
Thanks @rspurgeon!!  LGTM but would also appreciate @confluentinc/connect review for final approval.

C0urante commented:
Looks like the build is failing with some compilation errors:

```
[2020-09-24T23:46:41.804Z] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project kafka-connect-datagen: Compilation failure: Compilation failure: 
[2020-09-24T23:46:41.804Z] [ERROR] /home/jenkins/workspace/c-pr_kafka-connect-datagen_PR-71/src/main/java/io/confluent/kafka/connect/datagen/DatagenConnectorConfig.java:[19,33] package com.google.common.collect does not exist
[2020-09-24T23:46:41.804Z] [ERROR] /home/jenkins/workspace/c-pr_kafka-connect-datagen_PR-71/src/main/java/io/confluent/kafka/connect/datagen/DatagenConnector.java:[19,37] package com.google.common.annotations does not exist
[2020-09-24T23:46:41.804Z] [ERROR] /home/jenkins/workspace/c-pr_kafka-connect-datagen_PR-71/src/main/java/io/confluent/kafka/connect/datagen/DatagenConnector.java:[41,4] cannot find symbol
[2020-09-24T23:46:41.804Z] [ERROR]   symbol:   class VisibleForTesting
[2020-09-24T23:46:41.804Z] [ERROR]   location: class io.confluent.kafka.connect.datagen.DatagenConnector
[2020-09-24T23:46:41.804Z] [ERROR] /home/jenkins/workspace/c-pr_kafka-connect-datagen_PR-71/src/main/java/io/confluent/kafka/connect/datagen/DatagenConnectorConfig.java:[103,12] cannot find symbol
[2020-09-24T23:46:41.804Z] [ERROR]   symbol:   variable ImmutableList
[2020-09-24T23:46:41.804Z] [ERROR]   location: class io.confluent.kafka.connect.datagen.DatagenConnectorConfig
```

Looks like it might be Guava-related?

rspurgeon commented:
> Looks like the build is failing with some compilation errors:
> 
> Looks like it might be Guava-related?

I'm not sure how to diagnose this, I just successfully built locally with:

```
mvn --batch-mode -Pjenkins clean verify install dependency:analyze site validate -U
...
...
...
...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  04:09 min
[INFO] Finished at: 2020-09-25T07:39:33-05:00
[INFO] ------------------------------------------------------------------------
```

C0urante commented:
@rspurgeon It looks like this branch isn't up-to-date with the latest master. When Jenkins builds, it first rebases this current branch with master, and it looks like that's causing the build failures.

Can you run `git rebase master` on this branch? That should allow you to reproduce and diagnose the build failures.

rspurgeon commented:
@C0urante rebased off branch, added guava dependency explicitly.  Also switched to builder interface in avro generator to allieviate build warning.  LMK what you think... thanks

C0urante commented on src/test/java/io/confluent/kafka/connect/datagen/DatagenTaskTest.java:
```diff
@@ -373,10 +373,11 @@ private void loadKeyAndValueSchemas(String schemaResourceFilename, String idFiel
-          getClass().getClassLoader().getResourceAsStream(schemaFilename),
-          new Random()
-      );
+      InputStream schemaStream = getClass().getClassLoader().getResourceAsStream(schemaFilename);
+      Generator generator = new Generator.Builder()
+              .schemaStream(schemaStream)
+              .random(new Random())
+              .generation(0L).build();
       return generator.schema();
```
I think I left this comment before when reviewing https://github.com/confluentinc/kafka-connect-datagen/pull/64; we don't need to construct a new `Generator` just to parse an Avro schema:
```suggestion
      InputStream schemaStream = getClass().getClassLoader().getResourceAsStream(schemaFilename);
      return org.apache.avro.Schema.Parser().parse(schemaStream);
```

rspurgeon replied:
I didn't understand that either, but I was passing by quickly...

C0urante reviewed (approved):
LGTM! I've left one stylistic suggestion but as long as this builds successfully on Jenkins, it can be addressed at your discretion @rspurgeon.

rspurgeon commented:
I'll fix the build error @C0urante

C0urante commented:
🤦‍♀️ thanks, sorry about that Rick.
